Sources/StreamChat/StateLayer/MessageSearchState.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+/// Represents a list of message search results.
+@available(iOS 13.0, *)
+public final class MessageSearchState: ObservableObject {
+    private var cancellables = Set<AnyCancellable>()
```

Comment:
Can you clarify how are these cancelables working? 🤔 Since they are private, how do we cancel them?

Reply:
This is only for internal observation for forwarding the query to the `Observer` object.

```swift
 $query
            .assign(to: \.query, on: observer)
            .store(in: &cancellables)
```